core/model/modx/sources/modfilemediasource.class.php

```diff
@@ -494,6 +494,21 @@ public function renameObject($oldPath,$newName) {
             return false;
         }
 
+        // Check that the new file has a file type that is allowed
+        $allowedFileTypes = explode(',',$this->xpdo->getOption('upload_files',null,''));
+        $allowedFileTypes = array_merge(explode(',',$this->xpdo->getOption('upload_images')),explode(',',$this->xpdo->getOption('upload_media')),explode(',',$this->xpdo->getOption('upload_flash')),$allowedFileTypes);
+        $allowedFileTypes = array_unique($allowedFileTypes);
+
+        $file_split = explode('.', $newName);
+        $ext = $file_split[count($file_split) - 1];
```

Comment:
Interesting way to get the extension - wouldn't it be easier to just use [pathinfo() with PATHINFO_EXTENSION](http://php.net/manual/en/function.pathinfo.php)?